src/plugins/sails.js

```diff
@@ -198,9 +211,14 @@ io.socket.on('leftGame', function (evData) {
 io.socket.on('connect', () => {
   // Request latest game state if socket reconnects during game
   const { username } = store.state.auth;
-  const inGame = router.currentRoute.value.name === ROUTE_NAME_GAME;
-  if (!inGame) {
-    return;
+  switch (router.currentRoute.value.name) {
+    case ROUTE_NAME_GAME:
+      return store.dispatch('requestReauthenticate', { username });
+    case ROUTE_NAME_SPECTATE: {
+      const gameId = Number(router.currentRoute.value.params.gameId);
```

Comment:
You might need to be defensive here. Just because you're coercing it doesn't mean it's valid.

Reply:
How should we handle an invalid gameId? You kinda need to go to the homepage at that point, right? So maybe a dialog that explains the problem with a button to go home? Otherwise could be an error snackbar and then going home

Reply:
I had a solution for this at one point. It should just be an auto direct to the homepage with a snack bar message (eventually) that it’s an invalid game id.